frontend/src/resources/rbac.ts

```diff
@@ -84,6 +111,47 @@ export interface ClusterRoleBinding {
   }
 }
 
+export interface Role {
+  apiVersion: RbacApiVersionType
+  kind: RoleKindType
+  metadata: Metadata
+  rules: {
+    verbs: string[]
+    apiGroups: string[]
+    resources: string[]
+    resourceNames: string[]
```

Comment:
```suggestion
    resourceNames?: string[]
```

Reply:
from https://kubernetes.io/docs/reference/kubernetes-api/authorization-resources/role-v1/#Role

> ResourceNames is an **optional** white list of names...

Reply:
I would anyway create a new `PolicyRule` interface at `frontend/src/resources/kubernetes-client.ts` containing

```
import { ObjectReference } from '@openshift-console/dynamic-plugin-sdk'

export interface LocalObjectReference {
  verbs: ObjectReference[]
  apiGroups: string[]
  resources: string[]
  resourceNames?: string[]
}
```

and I would also replace `ClusterRole.rules` from the same file